Sources/Lifecycle/Lifecycle.swift

```diff
@@ -83,61 +83,66 @@ public struct LifecycleHandler {
 
 public struct ServiceLifecycle {
     private let configuration: Configuration
-    private let lifecycle: ComponentLifecycle
 
-    /// Creates a `ComponentLifecycle` instance.
+    /// The internal `ComponentLifecycle` instance
+    ///
+    /// Designed for composition purposes, mainly for frameworks that need to offer both top-level strart/stop functionality and composition into larger systems.
+    /// In other words, should not be in used outside the context of building an Application framework.
+    public let guts: ComponentLifecycle
```

Comment:
I suggest `_lifecycle` instead of `guts` but not a strong opinion.

Reply:
this is totally open to opinions

Reply:
Yeah... guts is a bit weird :) 
Yim's `_lifecycle` suggestion is good, or `underlying` which I use often as well?

Reply:
`_` I guess would imply that it's not "stable" so perhaps `underlying` or `componentLifecycle` after all?
I suppose this is effectively stable API after all right?

Reply:
If I read the code correctly, having it public will mean we allow multiple `ServiceLifecycle` instances to be initialised in the application simultaneously. Would it lead to multiple backtraces signal handlers being set up? I see it's a default behaviour for `ServiceLifecycle`.

Reply:
@ktoso @yim-lee `underlying` works for me

Reply:
@avolokhov the goal here is not to allow multiple ServiceLifecycle instances to be initialised in the application simultaneously, the goal is to enable composition, see [exchange with @tanner0101](https://forums.swift.org/t/sswg-0015-swift-service-lifecycle/39157/6?u=tomerd) for background

Reply:
Thanks for the context! Sorry if I'm about to talk nonsense, I'm trying to better understand the use case.
According to the sample in that thread
```
let lifecycle = ServiceLifecycle()

let app = try Application(.detect())
// app.lifecycle is ComponentLifecycle
lifecycle.register(app.lifecycle)

// other stuff

try lifecycle.startAndWait()
```
`app.lifecycle` here is a `ComponentLifecycle`, which is also `guts` (or `underlying`) for some `ServiceLifecycle` defined inside `Application`? In this case, `Application` doesn't know in advance whether it'll be used as a part of a bigger `Lifecycle` or as a top level app, so it'll have to install backtraces when creating its own `ServiceLifecycle`.
Then if it's used as a part of a bigger `Lifecycle` this bigger `Lifecycle` will have to install backtraces on its own. This will lead to multiple backtraces handlers set up in a single application. If I read it correctly, to avoid it we'll have to install backtraces handler on `start`, rather than on `init`?

I tried to follow the case @tanner0101 described in that thread https://forums.swift.org/t/sswg-0015-swift-service-lifecycle/39157/7
And got the following sample code:

```
    struct Application {
        let appLifecycle: ServiceLifecycle
        init() {
            appLifecycle = ServiceLifecycle()
        }

        var lifecycle: ComponentLifecycle {
            appLifecycle.underlying
        }

        func start() {
            appLifecycle.start {_ in }
        }

        func shutdown() {
            appLifecycle.shutdown()
        }
    }



    func testPartOfBiggerApp() throws {
        let lifecycle = ServiceLifecycle()

        let app = try Application()
        // app.lifecycle is ComponentLifecycle
        lifecycle.register(app.lifecycle)

        // other stuff

        try lifecycle.startAndWait()
    }
```
In this sample code two backtrace handlers are installed: one in `Application. appLifecycle.init()`, the other in `testPartOfBiggerApp` in `lifecycle.init()`. In general, my understanding is that a call to `lifecycle.underlying` implies that this `lifecycle` is created, and therefore backtrace handler is installed. When we use it as a `LifecycleTask` for some other `ServiceLifecycle`, we assume that the other `ServiceLifecycle` exists and therefore installed backtrace handler again.

Reply:
@avolokhov great catch, this is in fact a problem with the proposed solution 

I think I will modify this PR to address the other small improvements it includes but revert `underlying` to be private until we have a better solution

cc @tanner0101

Reply:
> If I read it correctly, to avoid it we'll have to install backtraces handler on start, rather than on init?

this could be one solution yes. the reason we had it in `init` originally was to install them as soon as possible as the application may crash before calling start (from real-world experience)

Reply:
another solution could be for application frameworks like vapor to take in an optional `Lifecycle` in their constructor

Reply:
@avolokhov finally got back to this PR... updated to moving the backtrace installer to `start` as you suggested